physionet-django/project/modelcomponents/activeproject.py

```diff
@@ -610,14 +610,12 @@ def is_editable_by(self, user):
         """
         Whether the user can edit the project
         """
-        if not user.has_perm('project.change_activeproject'):
-            project_authors = [author.user for author in self.author_list()]
-            if user in project_authors:
-                if user != self.submitting_author().user or not self.author_editable():
-                    return False
-            elif user == self.editor:
-                if not self.copyeditable():
-                    return False
-            else:
+        if user == self.submitting_author().user:
+            if not self.author_editable():
+                return False
+        elif user == self.editor:
+            if not self.copyeditable():
                 return False
+        else:
+            return False
         return True
```

Comment:
Can this final `return True` be removed? Doesn't look like it would ever get called.

Reply:
Ah that's right. How stupid i missed that. As backup, normally IDE would highlight stuff like this.

let me push the change.

Reply:
let me check again, it might be reachable now that i look at it. and probably needed

Reply:
@tompollard  I think we need it as the logic is that

If the user is  submitting author and project is on the right stage ---> allow edit
or if the user is editor and project is on right stage ---> allow edit
otherwise dont allow edit


and the block of code should return False if the right conditions are not met. and if all conditions are met return true.
I think if i refactor the func like this, it will be more readable

```
        if user == self.submitting_author().user:
            if self.author_editable():
                return True
        elif user == self.editor:
            if self.copyeditable():
                return True
        return False
```

or 

```
        if user == self.submitting_author().user:
            if self.author_editable():
                return True
        elif user == self.editor:
            if self.copyeditable():
                return True
        else:
            return False
```

Reply:
Let me know what you think. i can submit the change